scripts/models/event.js

```diff
@@ -88,34 +88,36 @@
         // TODO:  Once data is sanatized use return TownHall.filters[key].indexOf(townhall[key]) !== -1;
         return TownHall.filters[key].some(function(filter) {
           return filter.slice(0, 8) === townhall[key].slice(0, 8);
-        })
-      })
+        });
+      });
     }, data).sort(TownHall.sortFunction);
-  }
+  };
 
   // METHODS IN RESPONSE TO lookup
   // Converts zip to lat lng google obj
   TownHall.lookupZip = function (zip) {
-    return firebasedb.ref('/zips/' + zip).once('value').then(function(snapshot) {
-      var representativePromise = $.ajax({
-        url: 'https://congress.api.sunlightfoundation.com/legislators/locate?zip=' + zip,
-        dataType: 'jsonp'
-      });
-      var zipQueryLoc = new google.maps.LatLng(snapshot.val().LAT, snapshot.val().LNG);
-      TownHall.zipQuery = zipQueryLoc;
-      TownHall.returnNearest(zipQueryLoc).then(function(sorted) {
-        eventHandler.render(sorted, zipQueryLoc, representativePromise);
+    return new Promise(function (resolve, reject) {
+      firebasedb.ref('/zips/' + zip).once('value').then(function(snapshot) {
```

Comment:
I'm curious what "snapshot" refers to in this line.  Is there some kind of local caching that Firebase is doing?  So it's just the value returned (if there is one)?  Again, Firebase newb.